src/Program.cs

```diff
@@ -0,0 +1,461 @@
+                }
+                subidentexit = false;
+                subidentvalue = ParserGrammar.subidentvalue.Parse(line);
+            }
+        }
+        private void getConfigureIdent(string line,out bool configureexit, out string ident)
+        {
+            ident = String.Empty;
+            configureexit = true;
+
+            if (ParserGrammar.getidentifier.Parse(line) == "configure network" || ParserGrammar.getidentifier.Parse(line) == "configure voip")
+            {
+                ident = ParserGrammar.getidentifier.Parse(line);
+                configureexit = false;
+            }
+        }
+        public ACConfig parseinobject(string path)
+        {
+
+
+            Configureviop vo = new Configureviop();
+            ConfigureNetwork co = new ConfigureNetwork();
+            ACConfig AC = new ACConfig();
+            AC.configureNetwork = co;
+            AC.configureviop = vo;
+
+            List<Networkdev> netlist = new List<Networkdev>();
+            List<Interfacenetworkif> inif = new List<Interfacenetworkif>();
+            List<Proxyip> prip = new List<Proxyip>();
+            List<Proxyset> prese = new List<Proxyset>();
+            
+
+            string ident = " ";
+            bool configureexit = true;
+            string subident = " ";
+            bool subidentexit = true;
+            string subidentvalue = "";
+
+            bool activate = false;
+
+            string underlying = "";
+            string name = "";
+            string tagging = "";
+
+            string apptype = "";
+            string ipaddr = "";
+            int prel = 0;
+            string gateway = "";
+            string name2 = "";
+            string udev = "";
+            int vlan = 0;
+
+            string prname = "";
+            string peka = "";
+            string srdname = "";
+            string ssin = "";
+            string kfr = "";
+            int sdr = 0;
+            int sdi = 0;
+            string prm = "";
+            int iphs = 0;
+            int plbm = 0;
+            int masl = 0;
+
+            string prad = "";
+            string taty = "";
+
+
+            using (StreamReader Reader = new StreamReader(path))
+            {
+                string line = " ";
+                while ((line = Reader.ReadLine()) != null)
+                {
+                    if (configureexit)
+                    {
+                        getConfigureIdent(line, out configureexit, out ident);
+                        continue;
+                    }
+                    if (subidentexit)
+                    {
+                        getIdentNameAndValue(line, configureexit, subidentexit, out subident, out subidentvalue);
+                        continue;
+                    }
+
+                    if (configureexit == false && subidentexit == false && ident == "configure network" && subident == "network-dev")
+                    {
+                        switch (ParserGrammar.dev.Parse(line))
```

Comment:
hierfür brauchst du kein Switch case. Das hier kannst du viel besser lösen wenn du einen allgemeinen parser für die Value hast der den Namen als string entgegennimmt und die value zurückgibt.

Also ca. so:

```csharp
var value = ParserGrammer.value(ParserGrammar.dev.Parse(line)).Parser(line);
```

Reply:
aber alles was du hier machst sollte unbedingt auch in deine Grammatik mit rein. Lies dazu mal die Kapitel `Parsing the Question` und `Parsing a Section` (hier)[https://nblumhardt.com/2010/01/building-an-external-dsl-in-c/]

Reply:
weil im endeffekt kannst du diese ganze Methoder vermutlich in eine Zeile Auslagern:
`ParserGrammer.AcConfig.Parse(Reader)`